pandas/core/dtypes/concat.py

```diff
@@ -122,7 +122,11 @@ def is_nonempty(x) -> bool:
     any_ea = any(is_extension_array_dtype(x.dtype) for x in to_concat)
 
     if any_ea and axis == 1:
-        to_concat = [np.atleast_2d(x.astype("object")) for x in to_concat]
+        if single_dtype and ignore_2d_ea:
```

Comment:
can you mark this with a "# TODO(EA2D): ..."

Reply:
I plan to look into the motivation for the atleast_2d first. It seems strange to me that `concat_compat` would be concerned with reshaping, rather than just concatenation. I would think that the caller would take care of that.